clang/lib/Sema/CheckExprLifetime.cpp

```diff
@@ -582,6 +582,15 @@ static void visitFunctionCallArguments(IndirectLocalPath &Path, Expr *Call,
     //   Temp().ptr; // Here ptr might not dangle.
     if (isa<MemberExpr>(Arg->IgnoreImpCasts()))
       return;
+    // Avoid false positives when the object is constructed from a conditional
+    // operator argument. A common case is:
+    //   // 'ptr' might not be owned by the Owner object.
+    //   std::string_view s = cond() ? Owner().ptr : sv;
```

Comment:
I am a bit confused to be honest. Is there any other contexts where `Owner().ptr` is problematic? I'd expect our analysis to behave the same for a subexpression like that regardless the context. So I am surprised we need to insert special logic for the ternary operator.

Reply:
> I am a bit confused to be honest. Are there any other contexts where `Owner().ptr` is problematic? I'd expect our analysis to behave the same for a subexpression like that regardless the context. So I am surprised we need to insert special logic for the ternary operator.

Possibly, but I don’t know of any concrete examples (it’s hard to judge). We have another ad-hoc filter at the end, `IsGslPtrValueFromGslTempOwner`, which filters out cases where the GSL pointer doesn’t originate from a GSL owner. This works well for simple and common cases, but when combined with `lifetimebound`, the behavior becomes tricky. 

The current fix extends the `MemberExpr` logic (L583) to handle cases like `GSLPointer pointer(Owner().ptr);`, but it doesn’t yet address cases like `GSLPointer pointer(Cond ? Owner().ptr : GSLPointer());`. I think this fix is a reasonable extension to address the issue.

Reply:
> I am wondering if it was possible to use the same or similar MemberExpr filter when we drill down to the branches of the conditional operator. If it is too hard to do, 

This was my first approach. We could add a special `MemberExpr` filter inside the `do-while` loop in `visitLocalsRetainedByReferenceBinding`, but I don’t think it’s a good idea:

- the filtering logic is GSL-pointer specific, and `visitLocalsRetainedByReferenceBinding` shouldn’t be aware of it;  (layering violation)
- adding such filtering causes new false negatives in cases like `const string_view& sv = Owner().sv;`.  
- it would scatter the special filtering logic into another place, making the code harder to reason about.

Reply:
Is `const string_view& sv = Owner().sv;` a false negative? We cannot actually know if `sv` is actually owned or not because we have no lifetimebound annotation and the gsl owner does not tell us anything about fields, only about conversions/ctors.

Reply:
Ah, never mind, I missed the const ref!

Reply:
OK, with all of these being said I am OK with the current approach.